frontend/app/(dashboard)/questions/create/page.tsx

```diff
@@ -0,0 +1,128 @@
+      if (error.response && error.response.status === 409) {
+        alert("A question with this title already exists.");
+      } else {
+        alert("Failed to create question. Please try again.");
+      }
+    }
+  };
+
+  return (
+    <Box p={8} h="100%">
+      <FormControl id="title" mb={4} isRequired>
+        <FormLabel>Title</FormLabel>
+        <Input name="title" value={title} onChange={handleTitleChange} />
+      </FormControl>
+
+      <Flex direction={{ base: 'column', md: 'row' }} gap={4} mb={4}>
+        <FormControl id="description" isRequired flex="1">
+          <FormLabel>Description</FormLabel>
+          <Textarea
+            name="description"
+            value={description}
+            onChange={handleDescriptionChange}
+            minHeight={200} // some weird bug where setting height={200} = height 200+ idk why
+            maxHeight={200}
+            resize={"none"}
+          />
+        </FormControl>
+
+        <Box display="flex" flexDirection="column" flex="1">
+          <FormLabel>Preview</FormLabel>
+          <Box p={4} border="1px" borderColor="gray.200" borderRadius="md" flex="1" overflow="auto" minHeight={200} maxHeight={200}>
+            <div dangerouslySetInnerHTML={{ __html: marked(description) }}></div>
+          </Box>
+        </Box>
+      </Flex>
+
+      <FormControl id="topics" mb={4} isRequired>
+        <FormLabel>Topics</FormLabel>
+        <Select name="topics" value="" onChange={handleTopicsChange}>
+          <option value="" disabled>Select a topic</option>
+          {Object.values(QuestionTopic).map((topic) => (
+            <option key={topic} value={topic}>
+              {topic}
+            </option>
+          ))}
+        </Select>
+        <Box mt={4}>
+          {topics.size > 0 ? (
+            Array.from(topics).map((topic: QuestionTopic, idx: number) => (
+              topicText(topic, idx, () => {
+                const updatedTopics = new Set(topics);
+                updatedTopics.delete(topic);
+                setTopics(updatedTopics);
+              })
+            ))
+          ) : (
+            <Box>No topics selected</Box>
+          )}
+        </Box>
+      </FormControl>
+
+      <FormControl id="complexity" mb={4} isRequired>
+        <FormLabel>Complexity</FormLabel>
+        <Select name="complexity" value={complexity || ""} onChange={handleComplexityChange}>
+          <option value="" disabled>Select complexity</option>
+          <option value={QuestionComplexity.EASY}>Easy</option>
+          <option value={QuestionComplexity.MEDIUM}>Medium</option>
+          <option value={QuestionComplexity.HARD}>Hard</option>
+        </Select>
+      </FormControl>
+
+      <FormControl id="link" mb={4} isRequired>
+        <FormLabel>Link</FormLabel>
+        <Input name="link" value={link} onChange={handleLinkChange} />
+      </FormControl>
+
+      <Button colorScheme="teal" onClick={handleSubmit}>
+        Create
+      </Button>
```

Comment:
Actually would it be good to have a 'cancel' button that navigates the the previous page?

Reply:
Good suggestion, added cancel button

<img width="215" alt="image" src="https://github.com/user-attachments/assets/d77c9af2-f062-42e3-99f9-3949e1de25d8">